common/serviceerror/invalid_dispatch_build_id.go

```diff
@@ -0,0 +1,70 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package serviceerror
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.temporal.io/server/api/errordetails/v1"
+)
+
+type (
+	// InvalidDispatchBuildId happens when matching wants to dispatch task to a wrong build id
+	InvalidDispatchBuildId struct {
+		Message string
+		st      *status.Status
+	}
+)
+
+func NewInvalidDispatchBuildId() error {
+	return &InvalidDispatchBuildId{
+		Message: "dispatch build id is not the workflows current build id",
+	}
+}
+
+// Error returns string message.
+func (e *InvalidDispatchBuildId) Error() string {
+	return e.Message
+}
+
+func (e *InvalidDispatchBuildId) Status() *status.Status {
+	if e.st != nil {
+		return e.st
+	}
+
+	st := status.New(codes.Unavailable, e.Message)
+	st, _ = st.WithDetails(
+		&errordetails.StickyWorkerUnavailableFailure{},
+	)
```

Comment:
Why `Unavailable`?
Why reuse `StickyWorkerUnavailableFailure`?

I believe there are good reasons but could you please document those in the code.